File: func_log/zadania1_5/zadanie1.js
// нефункциональный

function reverseStringImperative(s) {
    let result = "";
    for (let i = s.length - 1; i >= 0; i--) {
        result += s[i];
    }
    return result;
}

console.log(reverseStringImperative("Hello"));  // olleH


// функциональный

const reverseStringFunctional = s => 
    s.split('').reverse().join('');

console.log(reverseStringFunctional("Hello"));  // olleH

File: func_log/zadania1_5/zadanie1.py
# нефунциональный 

def reverse_string_imperative(s):
    result = ""
    for i in range(len(s) - 1, -1, -1):
        result += s[i]
    return result

print(reverse_string_imperative("Hello"))  # olleH


#функциональный

def reverse_string_functional(s):
    return "".join(reversed(s))

print(reverse_string_functional("Hello"))  # olleH


